addons/stock_landed_costs/models/stock_landed_costs.py

```diff
@@ -0,0 +1,273 @@
+    date = fields.Date(required=True, states={'done': [('readonly', True)]}, default=fields.Date.context_today, track_visibility='onchange', copy=False)
+    picking_ids = fields.Many2many('stock.picking', string='Pickings', states={'done': [('readonly', True)]}, copy=False)
+    cost_line_ids = fields.One2many('stock.landed.cost.lines', 'cost_id', string='Cost Lines', states={'done': [('readonly', True)]})
+    valuation_adjustment_line_ids = fields.One2many('stock.valuation.adjustment.lines', 'cost_id', string='Valuation Adjustments', states={'done': [('readonly', True)]})
+    description = fields.Text(string='Item Description', states={'done': [('readonly', True)]})
+    amount_total = fields.Float(compute='_compute_amount_total', string='Total', digits=dp.get_precision('Account'), store=True,
+                                track_visibility='always')
+    state = fields.Selection([('draft', 'Draft'), ('done', 'Posted'), ('cancel', 'Cancelled')], default='draft', readonly=True, track_visibility='onchange', copy=False)
+    account_move_id = fields.Many2one('account.move', string='Journal Entry', readonly=True, copy=False)
+    account_journal_id = fields.Many2one('account.journal', string='Account Journal', required=True)
+
+    @api.depends('cost_line_ids', 'cost_line_ids.price_unit', 'cost_line_ids.cost_id')
+    def _compute_amount_total(self):
+        for landed_cost_line in self:
+            landed_cost_line.amount_total = sum(landed_cost_line.cost_line_ids.mapped('price_unit'))
+
+    def get_valuation_lines(self):
+        lines = []
+        if not self.picking_ids:
+            return lines
+
+        for picking in self.picking_ids:
+            for move in picking.move_lines.filtered(lambda move: move.product_id.valuation == 'real_time' or move.product_id.cost_method == 'real'):
+                #it doesn't make sense to make a landed cost for a product that isn't set as being valuated in real time at real cost
+                weight = move.product_id.weight * move.product_qty
+                volume = move.product_id.volume * move.product_qty
+                total_cost = sum([quant.cost * quant.qty for quant in move.quant_ids])
+                vals = dict(product_id=move.product_id.id, move_id=move.id, quantity=move.product_uom_qty, former_cost=total_cost, weight=weight, volume=volume)
+                lines.append(vals)
+        if not lines:
+            raise UserError(_('The selected picking does not contain any move that would be impacted by landed costs. Landed costs are only possible for products configured in real time valuation with real price costing method. Please make sure it is the case, or you selected the correct picking'))
+        return lines
+
+    def _create_accounting_entries(self, line, move, qty_out):
+        cost_product = line.cost_line_id and line.cost_line_id.product_id
```

Comment:
No need to check the condition. Just cost_product = line.cost_line_id.product_id is enough .